diff --git a/ftplugin/quicktask.vim b/ftplugin/quicktask.vim
index 15dd0cf..17acbcf 100755
--- a/ftplugin/quicktask.vim
+++ b/ftplugin/quicktask.vim
@@ -1069,7 +1069,7 @@ if ! g:quicktask_no_mappings && ! exists('b:quicktask_did_mappings')
     nmap <unique><buffer> <Leader>td  <Plug>MoveTaskDown
     nmap <unique><buffer> <Leader>tS  <Plug>AddSnipToTask
     nmap <unique><buffer> <Leader>tfi <Plug>FindIncompleteTimestamps
-    nmap <unique><buffer> <CR>        <Plug>OpenSnipUnderCursor
+    nmap <buffer> <CR>        <Plug>OpenSnipUnderCursor
     let b:quicktask_did_mappings = 1
 endif
 
diff --git a/plugin/syntastic.vim b/plugin/syntastic.vim
index 902e7ba..05c453d 100644
--- a/plugin/syntastic.vim
+++ b/plugin/syntastic.vim
@@ -558,7 +558,7 @@ endfunction " }}}2
 function! s:skipFile() " {{{2
     let fname = expand('%')
     let skip = (exists('b:syntastic_skip_checks') ? b:syntastic_skip_checks : 0) ||
-        \ (&buftype != '') || !filereadable(fname) || getwinvar(0, '&diff') ||
+        \ (&buftype != '') || !filereadable(fname) ||
         \ s:ignoreFile(fname) || fnamemodify(fname, ':e') =~? g:syntastic_ignore_extensions
     if skip
         call syntastic#log#debug(g:SyntasticDebugTrace, 'skipFile: skipping')
diff --git a/autoload/neocomplete/sources/ultisnips.vim b/autoload/neocomplete/sources/ultisnips.vim
index 385a779..44eee4e 100644
--- a/autoload/neocomplete/sources/ultisnips.vim
+++ b/autoload/neocomplete/sources/ultisnips.vim
@@ -9,7 +9,7 @@ let s:source = {
    \ 'matchers' :
       \ (g:neocomplete#enable_fuzzy_completion ?
       \ ['matcher_fuzzy'] : ['matcher_head']),
-   \ 'min_pattern_length' : 1,
+   \ 'min_pattern_length' : 4,
    \ 'max_candidates' : 20,
    \ 'is_volatile': 1,
    \ }
